packages/flame/test/sprite_animation_test.dart

```diff
@@ -90,6 +92,27 @@ void main() {
       expect(counter, 1);
     });
 
+    test(
+        'verify call is being made at the first of the frame for multi-frame animation',
+        () {
+      var timePassed = 0.0;
+      const dt = 0.5;
+      var timesCalled = 0;
+      final sprite = MockSprite();
+      final spriteList = [sprite, sprite, sprite];
+      final animation =
+          SpriteAnimation.spriteList(spriteList, stepTime: 1, loop: false);
+      animation.onFrame = (index) {
+        expect(index, timePassed);
+        timesCalled++;
+      };
+      while (timePassed <= 3) {
+        timePassed += dt;
+        animation.update(dt);
+      }
+      expect(timesCalled, spriteList.length - 1);
```

Comment:
This should be `spriteList.length`, right?

Reply:
No, because its 0 based, onFrame never moves timesCalled above 2 (0,1,2), so we need length -1.

Reply:
`timesCalled` is the number of times `onFrame()` was called, because it's incremented inside `onFrame()`. So it's supposed to be equal 3 here.

Reply:
But its starting at 0 on the first call of onFrame.  0 increments to 1 for the second call of onFrame.  1 increments to 2 on the third call of onFrame.  I have stepped through this with breakpoints to verify that onFrame gets called 3 times.

Reply:
Since it is incrementing the number once per call, it should be 3, since there are 3 frames. 
It is incremented to 1 on the first call, not to 0.

Reply:
Ok, ever since we switched things around, the third frame does not get called because the animation has loop false.  This is what I was asking about earlier.  Because its the last frame, it does not call onFrame for the last frame.  I know you said that onComplete doesn't get called until the end, but I am not certain of that, because onFrame does not get called the third time.

Reply:
If you set a break point on the test on lines 107 and 111, you can watch timePassed increment above 3, but it does not call onFrame.  onComplete does fire though.

Reply:
> Ok, ever since we switched things around, the third frame does not get called because the animation has loop false.  This is what I was asking about earlier.  Because its the last frame, it does not call onFrame for the last frame.  I know you said that onComplete doesn't get called until the end, but I am not certain of that, because onFrame does not get called the third time.

It is not the last frame that is skipped, it's the first. We need to call the `onFrame` callback at the same time as `onStart` too.

> So to clarify, when it is last frame and loop=false, onFrame does not get called for the last frame.  Is that expected behaviour?  

It is not expected behavior, it should be called for each frame.

Reply:
Ok, so will do

Reply:
Ok, all verified being called correctly now.  Yeah, sorry, I don't know why it was taking me so long to figure this out.

Reply:
Also, I removed one test because this test now covers number of frames process = number of frames in the animation.